src/memory.cc

```diff
@@ -97,14 +97,20 @@ void* operator new(size_t size) BADALLOC
 		}
 		else
 		{
+#if defined(_CXXRT_NO_EXCEPTIONS)
+			break;
+#else
 			throw std::bad_alloc();
+#endif
```

Comment:
Is returning `nullptr` the right behaviour here?  Callers of this will never perform a `nullptr` check because the contract for the may-throw version is that it will never fail and return on the no-exception path.  I'd prefer if this either called `abort` (we've already tried to call a callback that allowed the programmer to specify something else to handle this and we don't hit the code path if it exists).